validator/sawtooth_validator/execution/scheduler_parallel.py

```diff
@@ -593,6 +593,9 @@ def _has_predecessors(self, txn):
         for predecessor_id in self._txn_predecessors[txn.header_signature]:
             if predecessor_id not in self._txn_results:
                 return True
+            for pre_pred_id in self._txn_predecessors[predecessor_id]:
```

Comment:
Since the reason this change is needed seems fairly complex, a comment explaining why the check is being performed here seems like a good idea.

Reply:
Yeah, I am going to do some more testing to see if I can break this with a unit test.